src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1321,6 +1321,12 @@ class Simplex_tree {
         return Dit_value_t(v, Node(&root_, filt)); });
     root_.members_.insert(boost::begin(verts), boost::end(verts));
     if (dimension_ < 0 && !root_.members_.empty()) dimension_ = 0;
+    if constexpr (Options::link_nodes_by_label) {
+      for (auto sh = root_.members().begin(); sh != root_.members().end(); sh++) {
+        if (nodes_label_to_list_.find(sh->first) == nodes_label_to_list_.end())
```

Comment:
Or something like
```suggestion
        if (!sh->second.list_max_vertex_hook_.is_linked())
```
?